src/jdk.internal.le/share/classes/jdk/internal/org/jline/JdkConsoleProviderImpl.java

```diff
@@ -98,7 +98,7 @@ public JdkConsole printf(String format, Object ... args) {
         public String readLine(String fmt, Object ... args) {
             try {
                 initJLineIfNeeded();
-                return jline.readLine(fmt.formatted(args));
+                return jline.readLine(fmt.formatted(args).replace("%", "%%"));
```

Comment:
I understand that [JLine interprets `%` in a prompt](https://github.com/openjdk/jdk/blob/4ed38f5ad5f822ab948257ed39717ea919fd32ed/src/jdk.internal.le/share/classes/jdk/internal/org/jline/reader/impl/LineReaderImpl.java#L4050), but are the interpretation rules documented on JLine GitHub page or elsewhere?

Reply:
There's a description in the javadoc:
https://www.javadoc.io/doc/org.jline/jline/latest/org/jline/reader/LineReader.html